packages/assets-controllers/src/TokensController.test.ts

```diff
@@ -1033,95 +999,27 @@ describe('TokensController', () => {
           type: ApprovalType.WatchAsset,
           requestData: {
             id: requestId,
-            interactingAddress,
-            asset: {
-              address: asset.address,
-              decimals: asset.decimals,
-              symbol: asset.symbol,
-              image: asset.image,
-            },
+            interactingAddress: '0x1',
+            asset,
           },
         },
         true,
       );
 
-      generateRandomIdStub.restore();
-      clock.restore();
+      generateRandomIdStub.mockRestore();
     });
 
-    it.each([
-      ['resolves', true],
-      ['rejects', false],
-    ])(
-      'should add token correctly if user confirms and message to ApprovalController %s',
-      async function (_, approvalControllerCallResolves: boolean) {
-        const generateRandomIdStub = sinon
-          .stub(tokensController, '_generateRandomId')
-          .callsFake(() => requestId);
-        type = 'ERC20';
-
-        let calledOnce = false;
-        const callActionSpy = approvalControllerCallResolves
-          ? jest.spyOn(messenger, 'call').mockResolvedValue({})
-          : jest.spyOn(messenger, 'call').mockImplementation(() => {
-              if (!calledOnce) {
-                calledOnce = true;
-                return Promise.resolve({});
-              }
-
-              throw new Error();
-            });
-
-        await tokensController.watchAsset(asset, type);
-        await tokensController.acceptWatchAsset(requestId);
-
-        expect(tokensController.state.suggestedAssets).toStrictEqual([]);
-        expect(tokensController.state.tokens).toHaveLength(1);
-        expect(tokensController.state.tokens).toStrictEqual([
-          {
-            isERC721: false,
-            aggregators: [],
-            ...asset,
-            image: 'image',
-          },
-        ]);
-        expect(callActionSpy).toHaveBeenCalledTimes(1);
-        expect(callActionSpy).toHaveBeenCalledWith(
-          'ApprovalController:addRequest',
-          {
-            id: requestId,
-            origin: ORIGIN_METAMASK,
-            type: ApprovalType.WatchAsset,
-            requestData: {
-              id: requestId,
-              interactingAddress: '0x1',
-              asset: {
-                address: asset.address,
-                decimals: asset.decimals,
-                symbol: asset.symbol,
-                image: asset.image,
-              },
-            },
-          },
-          true,
-        );
-
-        generateRandomIdStub.restore();
-      },
-    );
-
-    it('should store token correctly under interacting address if user confirms', async function () {
-      const generateRandomIdStub = sinon
-        .stub(tokensController, '_generateRandomId')
-        .callsFake(() => requestId);
-      type = 'ERC20';
+    it('stores token correctly under interacting address if user confirms', async function () {
```

Comment:
I appreciate it's a negative test as it's verifying we don't have any error handling logic, but should we have a test to verify that the method throws if the promise from the messenger call rejects?

Reply:
I can't see what value that'll add as part of a unit test. It's a promise that is being awaited and that is not wrapped in a try/catch, so yes, it will throw.

In an e2e test we can check that, when rejected, it indeed throws a `userRejectedRequest` RPC error.

Reply:
The value would be it would assert our assumption that it's not catching the approval errors, what you've said concerning the code is correct but the unit test would enforce that requirement going forward and ensure it's not captured and handled in future without us being aware and explicitly changing our expectations.